record-and-playback/presentation/scripts/publish/presentation.rb

```diff
@@ -370,39 +351,40 @@ def svg_render_shape_poll(g, slide, shape)
   num_responders = shape[:num_responders]
   presentation = slide[:presentation]
 
-  json_file = "#{$process_dir}/poll_result#{poll_id}.json"
-  svg_file = "#{$process_dir}/presentation/#{presentation}/poll_result#{poll_id}.svg"
+  json_file = "#{@process_dir}/poll_result#{poll_id}.json"
+  svg_file = "#{@process_dir}/presentation/#{presentation}/poll_result#{poll_id}.svg"
 
   # Save the poll json to a temp file
-  IO.write(json_file, result)
+  File.open(json_file, 'w') { |f| f.write result }
   # Render the poll svg
-  ret = BigBlueButton.exec_ret('utils/gen_poll_svg', '-i', json_file, '-w', "#{width.round}", '-h', "#{height.round}", '-n', "#{num_responders}", '-o', svg_file)
-  raise "Failed to generate poll svg" if ret != 0
+  ret = BigBlueButton.exec_ret('utils/gen_poll_svg', '-i', json_file, '-w', width.round.to_s, '-h', height.round.to_s,
+                               '-n', num_responders.to_s, '-o', svg_file)
+  raise 'Failed to generate poll svg' if ret != 0
 
   # Poll image
   g << doc.create_element('image',
-          'xlink:href' => "presentation/#{presentation}/poll_result#{poll_id}.svg",
-          width: width, height: height, x: x, y: y)
+                          'xlink:href' => "presentation/#{presentation}/poll_result#{poll_id}.svg",
+                          width: width, height: height, x: x, y: y)
 end
 
 def svg_render_shape(canvas, slide, shape, image_id)
-  if shape[:in] == shape[:out]
+  shape_in = shape[:in]
+  shape_out = shape[:out]
+
+  if shape_in == shape_out
     BigBlueButton.logger.info("Draw #{shape[:shape_id]} Shape #{shape[:shape_unique_id]} is never shown (duration rounds to 0)")
     return
   end
 
-  if shape[:in] >= slide[:out] or
-      (!shape[:out].nil? and shape[:out] <= slide[:in])
+  if (shape_in >= slide[:out]) || (shape_out && (shape_out <= slide[:in]))
     BigBlueButton.logger.info("Draw #{shape[:shape_id]} Shape #{shape[:shape_unique_id]} is not visible during image time span")
     return
   end
 
-  BigBlueButton.logger.info("Draw #{shape[:shape_id]} Shape #{shape[:shape_unique_id]} Type #{shape[:type]} from #{shape[:in]} to #{shape[:out]} undo #{shape[:undo]}")
-
   doc = canvas.document
   g = doc.create_element('g',
-          id: "image#{image_id}-draw#{shape[:shape_id]}", class: 'shape',
-          timestamp: shape[:in], undo: (shape[:undo].nil? ? -1 : shape[:undo]))
+                         id: "image#{image_id}-draw#{shape[:shape_id]}", class: 'shape',
+                         timestamp: shape_in, undo: shape[:undo] || -1)
```

Comment:
Heads up - @gigate is working on refactoring the undo system to allow for _redo_ as a prerequisite to the Eraser functionality (reference https://github.com/bigbluebutton/bigbluebutton/pull/13987 )
Is there much left before this can merge (so that @gigate can write their changes on top)?